### PR TITLE
Separate beneficiary in IGP

### LINKED ISSUE #1793: IGP should send paid funds to a separate non-owner beneficiary
We expect the owner of the IGP to be cold, like a multisig, because it wields a lot of configuration power. The funds that are sent into the IGP shouldn't then go to the owner multisig, requiring multisig signers to distribute funds back into the key funder. Instead, we should allow the owner to set a beneficiary, which in our deployed IGPs will be the key funder address

### PR DESCRIPTION
**Pull request by tkporter**
### Description

Because the owner of the IGP is likely to be a multisig and we want to ultimately redistribute funds sent to the IGP back to our key funder, this adds a `beneficiary` address to the IGP that can claim IGP payments

Includes:
* Contract changes
* Deploy / checker / govern tooling

### Drive-by changes

none

### Related issues

- Fixes #1793 

### Backward compatibility

_Are these changes backward compatible?_

Contracts are yes

_Are there any infrastructure implications, e.g. changes that would prohibit deploying older commits using this infra tooling?_

Sorta - the changes to the checker / govern tooling assume new functions exist on the current IGP

### Testing

_What kind of testing have these changes undergone?_

Unit tests

### CODE REVIEW

File: solidity/contracts/igps/InterchainGasPaymaster.sol [hunk 1]
```diff
@@ -23,6 +23,9 @@ contract InterchainGasPaymaster is
     /// @notice Keyed by remote domain, the gas oracle to use for the domain.
     mapping(uint32 => IGasOracle) public gasOracles;
 
+    /// @notice The benficiary that can receive native tokens paid into this contract.
+    address public beneficiary;
```
Comment (nambrot) [inline solidity/contracts/igps/InterchainGasPaymaster.sol:27]: Just double checking that we know that this storage addition is safe

**Review by nambrot — APPROVED** [review]

File: solidity/contracts/igps/InterchainGasPaymaster.sol (hunk 1 above)
Comment (tkporter) [reply solidity/contracts/igps/InterchainGasPaymaster.sol:27]: yep purely additive

Comment (tkporter) [reply solidity/contracts/igps/InterchainGasPaymaster.sol:27]: prior to performing the contract upgrade I'll be sure to inspect the storage layout before & what it'll look like after to ensure it's only additive https://book.getfoundry.sh/reference/forge/forge-inspect?highlight=storage%20layout#examples

Comment (nambrot) [reply solidity/contracts/igps/InterchainGasPaymaster.sol:27]: Feel free to do https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/1742 if you want :P

**Comment by tkporter** [comment]
Waiting till the branches this PR depends on are merged before resolving conflicts to avoid spending too much time resolving conflicts if those other branches require changes

**Review by yorhodes — APPROVED** [review]

**Comment by tkporter** [comment]
And merging into https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/1765 ...

**Pull request merged**